src/sniffer.c

```diff
@@ -2155,701 +2189,861 @@ static void ShowTlsSecrets(SnifferSession* session)
+                }
             }
-        }
-    #endif
+        #endif
 
-        if (ret == 0) {
-            ret = wc_InitDhKey_ex(&dhKey, NULL, devId);
-            if (ret == 0)
-                keyInit = 1;
-        }
-        if (ret == 0) {
-    #ifdef HAVE_PUBLIC_FFDHE
-            ret = wc_DhSetKey(&dhKey,
-                (byte*)params->p, params->p_len,
-                (byte*)params->g, params->g_len);
-            p_len = params->p_len;
-    #else
-            ret = wc_DhSetNamedKey(&dhKey, ksInfo->named_group);
             if (ret == 0) {
-                privKeySz = wc_DhGetNamedKeyMinSize(ksInfo->named_group);
-                ret = wc_DhGetNamedKeyParamSize(ksInfo->named_group,
-                        &p_len, NULL, NULL);
+                ret = wc_InitDhKey_ex(&args->key->priv.dh.key, NULL, devId);
+                if (ret == 0) {
+                    args->key->type = WC_PK_TYPE_DH;
+                    args->key->initPriv = 1;
+                #ifdef WOLFSSL_ASYNC_CRYPT
+                    ret = wolfSSL_AsyncInit(ssl, &args->key->priv.dh.key.asyncDev,
+                        WC_ASYNC_FLAG_NONE);
+                #endif
+                }
+            }
+            if (ret == 0) {
+        #ifdef HAVE_PUBLIC_FFDHE
+                ret = wc_DhSetKey(&args->key->priv.dh.key,
+                    (byte*)params->p, params->p_len,
+                    (byte*)params->g, params->g_len);
+                args->key->priv.dh.pLen = params->p_len;
+        #else
+                ret = wc_DhSetNamedKey(&args->key->priv.dh.key, ksInfo->named_group);
+                if (ret == 0) {
+                    args->key->priv.dh.privKeySz = wc_DhGetNamedKeyMinSize(ksInfo->named_group);
+                    ret = wc_DhGetNamedKeyParamSize(ksInfo->named_group,
+                            &args->key->priv.dh.pLen, NULL, NULL);
+                }
+        #endif
+            }
+            if (ret == 0) {
+                idx = 0;
+                ret = wc_DhKeyDecode(args->keyBuf->buffer, &idx, &args->key->priv.dh.key,
+                    args->keyBuf->length);
+            }
+            if (ret == 0) {
+                ret = wc_DhExportKeyPair(&args->key->priv.dh.key,
+                    args->key->priv.dh.privKey, &args->key->priv.dh.privKeySz,
+                    NULL, NULL);
             }
-    #endif
-        }
-        if (ret == 0) {
-            ret = wc_DhKeyDecode(keyBuf->buffer, &idx, &dhKey,
-                keyBuf->length);
-        }
-        if (ret == 0) {
-            ret = wc_DhExportKeyPair(&dhKey, privKey, &privKeySz, NULL,
-                NULL);
         }
+    #endif /* !NO_DH && WOLFSSL_DH_EXTRA */
 
-    #if defined(WOLFSSL_STATIC_EPHEMERAL) && !defined(SINGLE_THREADED)
-        if (keyLocked) {
-            wc_UnLockMutex(&ctx->staticKELock);
-        }
-    #endif
+    #ifdef HAVE_ECC
+        /* Static ECC Key */
+        if (useCurveId >= 0 && args->keyBuf == NULL
+        #ifdef HAVE_CURVE25519
```

Comment:
Does defining `HAVE_CURVE25519` require `HAVE_ECC`?  Should this be outside the `HAVE_ECC` guard?